x11-terms/roxterm: Bump to 3.9.2

ekohl opened the pull request:
Closes: https://bugs.gentoo.org/581272
Closes: https://bugs.gentoo.org/718548
Package-Manager: Portage-2.3.89, Repoman-2.3.20

ekohl commented:
While testing this, I did run into https://github.com/realh/roxterm/issues/222. The patch is submitted as https://github.com/realh/roxterm/pull/223.

ekohl commented:
As indicated in https://bugs.gentoo.org/581272#c3, I am interested in becoming a proxy maintainer. First I want to dig in the odd character issue because it's pretty annoying. In https://github.com/realh/roxterm/issues/222#issuecomment-617449057 I have traced it to the script provided by `x11-libs/vte-0.58.3` and come up with a workaround but I'm going to see if a newer vte would solve it.

mgorny commented:
Then please don't be shy and add yourself to metadata.xml. You can also add a commit removing p.mask entry.

ekohl commented:
Will do. Is it ok to keep this open while I track down the issue with vte and then update this? In the mean time I'm testing this new version via my personal overlay.

ekohl commented:
Something like this? I added one commit to add myself as proxy maintainer which also removes The package.mask. The other was updated to depend on vanilla vte as well as update the metadata remote-id. I wasn't sure how to split up commits and conventions in the repo.

juippis commented on profiles/package.mask:
```diff
@@ -127,11 +127,6 @@ media-gfx/svg2rlg
 # release. Bug 718326. Removal whenever it becomes an issue.
 mail-filter/pyzor
 
-# Michał Górny <mgorny@gentoo.org> (2020-04-19)
```
Please split this to a 3rd commit.

juippis commented on x11-terms/roxterm/metadata.xml:
```diff
@@ -11,6 +11,6 @@
 
 	<upstream>
-		<remote-id type="sourceforge">roxterm</remote-id>
+		<remote-id type="github">realh/roxterm</remote-id>
```
You can do this edit in while changing metadata.xml.

juippis commented on x11-terms/roxterm/roxterm-3.9.2.ebuild:
```diff
@@ -0,0 +1,39 @@
+SLOT="1"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND=">=dev-libs/dbus-glib-0.100
```
Version restriction can be lifted.

ekohl replied:
What are the reasons for this? Isn't it good practice to at least state the oldest version if there is one?

juippis replied:
Then you should specify minimum requirements for every dep, not just one. It's just a lot cleaner and easier to maintain this way.

juippis commented on x11-terms/roxterm/roxterm-3.9.2.ebuild:
```diff
@@ -0,0 +1,39 @@
+	sys-apps/dbus
+	x11-libs/gtk+:3
+	x11-libs/vte:2.91[vanilla]
+	!x11-terms/roxterm:0"
```
This blocker can be removed, as I can see :0 hasn't existed in years.

juippis commented on x11-terms/roxterm/roxterm-3.9.2.ebuild:
```diff
@@ -0,0 +1,39 @@
+# slotted versions of roxterm need block one another due to file collisions. #565288
+DEPEND="${RDEPEND}
+	dev-libs/libxslt
+	virtual/pkgconfig"
```
This is BDEPEND.

ekohl commented:
Updated to address comments. Also updated it to the now released 3.9.3 and dropped PATCHES since https://github.com/realh/roxterm/pull/223 is included.

juippis commented:
You need to create a bugzilla account with that e-mail you just put to metadata.xml so all bugs get assigned to you. If you already have a bugzilla account, just switch it to that e-mail.

ekohl commented:
Updated my Bugzilla email address.
